app/leaderboard/[duration]/Leaderboard.tsx

```diff
@@ -50,26 +50,65 @@ export default function Leaderboard(props: Props) {
 
   const search = searchParams.get("search") || "";
 
-  const roles = searchParams.get("roles")?.split(",") || [];
-  const ordering = ORDERING_OPTIONS.find(
-    (option) => option.value === searchParams.get("ordering"),
-  ) || { value: "points", text: "Points" };
-  const isReversed = searchParams.get("isReversed") === "true";
+  const [roles, setRoles] = useState<string[]>(
+    searchParams.get("roles")?.split(",") || [],
+  );
+
+  const [ordering, setOrdering] = useState<SelectOption>(
+    ORDERING_OPTIONS.find(
+      (option) => option.value === searchParams.get("ordering"),
+    ) || { value: "points", text: "Points" },
+  );
+
+  const [resultSet, setResultSet] = useState(props.data);
+
+  const [isReversed, setIsReversed] = useState<boolean>(
+    searchParams.get("isReversed") === "true",
+  );
```

Comment:
let's not have a useState for this.

Reply:
@rithviknishad we need a usestate else when we update result set it won't get render on UI. correct me if i am wrong

Reply:
so far when we update the search / role / date range, it is getting updated right? how was that working then without the useState?

Reply:
@rithviknishad can you check updated thanks for your insights